src/MediaUploader.php

```diff
@@ -512,6 +539,15 @@ public function update(Media $media)
         return $dirty;
     }
 
+    public function verifyFile()
+    {
+        return [
+            $this->verifyFileSize($this->source->size()),
+            $this->verifyMimeType($this->source->mimeType()),
+            $this->verifyExtension($this->source->extension()),
+        ];
```

Comment:
I think method should just return `true` or throw an exception. With the exception of `VerifyDestination()`, these methods due not modify the input, they only return the value as a convenience, so there isn't really a reason for this method to return values. You could just call this near the top of the upload function then assign the values manually, instead of using array destructuring (never been a fan of that).

There are a number of other things that could be verified (e.g. verifySource, etc.).